### PR TITLE
fix #151 handle multiple wallets same currency

### LINKED ISSUE #151: handle multiple wallets same currency

### PR DESCRIPTION
**Pull request by xiaoxianma**
enhance logging of creating payment methods

### CODE REVIEW

File: api/crud.py [hunk 1]
```diff
@@ -174,6 +175,9 @@ async def update_invoice_payments(invoice, wallets, discounts, store, product, p
     for wallet_id in wallets:
         wallet = await models.Wallet.get(wallet_id)
         method = await create_payment_method(invoice, wallet, product, store, discounts, promocode)
+        logger.debug(
+            f"Payment method(wallet_id={wallet_id}),store={store.id},currency={wallet.currency} is added to invoice {invoice.id}"
```
Comment (MrNaif2018) [inline api/crud.py:179]:
```suggestion
            f"Payment method(wallet={wallet_id}, store={store.id}, currency={wallet.currency}) is added to invoice {invoice.id}"
```

Comment (MrNaif2018) [inline api/crud.py:179]:
```suggestion
            f"Payment method(wallet={wallet_id}, store={store.id}, currency={wallet.currency}) is added to invoice {invoice.id}"
```

File: api/crud.py [hunk 2]
```diff
@@ -193,23 +197,27 @@ async def invoice_add_related(item: models.Invoice):
     item.products = [product_id for product_id, in result if product_id]
     item.payments = []
     payment_methods = await models.PaymentMethod.query.where(models.PaymentMethod.invoice_id == item.id).gino.all()
+    currency_payment_methods = defaultdict(list)
     for method in payment_methods:
-        # TODO: multiple wallet same currency case
-        # TODO remove duplication
```
Comment (MrNaif2018) [inline api/crud.py:198]: By remove duplication I meant, see the `_create_payment_method` method, it contains the same code but misses name attribute.
Please fix this, maybe we can make a class method `format` or `json` (probably the latter) that will return json representation of a model. In general I think in the future we will need to abstract this more

File: api/crud.py [hunk 3]
```diff
@@ -193,23 +197,27 @@ async def invoice_add_related(item: models.Invoice):
...
-                "discount": method.discount,
-                "currency": method.currency,
-                "lightning": method.lightning,
-                "node_id": method.node_id,
-            }
-        )
+        currency_payment_methods[method.currency].append(method)
+    for currency, methods in currency_payment_methods.items():
+        for i, method in enumerate(methods, start=1):
+            name = method.currency if len(methods) == 1 else f"{currency} ({i})"
```
Comment (MrNaif2018) [inline api/crud.py:205]: I think we can apply all the formatting the frontend does right there, as name is no longer used as dictionary key.
if it is lightning and there are multiple currencies, it should look like so:
BTC (⚡) (1)
BTC (⚡) (2)

Or do you have any better ideas?

**Review by MrNaif2018 — CHANGES_REQUESTED** [review]
Thank you for your PR! Please take a look at review comments

File: api/models.py [hunk 4]
```diff
@@ -248,6 +249,17 @@ class PaymentMethod(db.Model):
     lightning = Column(Boolean(), default=False)
     node_id = Column(Text)
 
+    async def to_enhanced_dict(self):
```
Comment (xiaoxianma) [inline api/models.py:252]: cannot pick a good name

**Comment by MrNaif2018** [comment]
Retrigerring CI

**Review by MrNaif2018 — APPROVED** [review]
Thank you for your PR! Merging!

**Pull request merged**